include/kodlab_mjbots_sdk/rotations.h

```diff
@@ -0,0 +1,410 @@
+   *
+   * @param rpy_in vector of roll, pitch, and yaw angles
+   */
+  explicit EulerAngles(const Eigen::Vector3<Scalar> &rpy_in)
+      : roll(rpy_in(0)), pitch(rpy_in(1)), yaw(rpy_in(2)) {}
+
+  /**
+   * @brief Construct an Euler Angles object with default convention.
+   *
+   * @param roll_in roll angle
+   * @param pitch_in pitch angle
+   * @param yaw_in yaw angle
+   */
+  EulerAngles(const Scalar &roll_in,
+              const Scalar &pitch_in,
+              const Scalar &yaw_in)
+      : roll(roll_in), pitch(pitch_in), yaw(yaw_in) {}
+
+  /**
+   * @brief Construct an Euler Angles object.
+   *
+   * @param roll_in roll angle
+   * @param pitch_in pitch angle
+   * @param yaw_in yaw angle
+   * @param axes_in array of rotation axes identifiers
+   * @param rot_type_in rotation type (Extrinsic or Intrinsic)
+   */
+  EulerAngles(const Scalar &roll_in,
+              const Scalar &pitch_in,
+              const Scalar &yaw_in,
+              const std::array<int, 3> &axes_in,
+              const EulerRotationType &rot_type_in)
+      : roll(roll_in), pitch(pitch_in), yaw(yaw_in), axes(axes_in),
+        rot_type(rot_type_in) {}
+
+  /**
+   * @brief Construct an Euler Angles object.
+   *
+   * @param rpy_in array of roll, pitch, and yaw angles
+   * @param axes_in array of rotation axes identifiers
+   * @param rot_type_in rotation type (Extrinsic or Intrinsic)
+   */
+  EulerAngles(const std::array<Scalar, 3> &rpy_in,
+              const std::array<int, 3> &axes_in,
+              const EulerRotationType &rot_type_in)
+      : roll(rpy_in[0]), pitch(rpy_in[1]), yaw(rpy_in[2]), axes(axes_in),
+        rot_type(rot_type_in) {}
+
+  /**
+   * @brief Construct an Euler Angles object.
+   *
+   * @param rpy_in vector of roll, pitch, and yaw angles
+   * @param axes_in array of rotation axes identifiers
+   * @param rot_type_in rotation type (Extrinsic or Intrinsic)
+   */
+  EulerAngles(const Eigen::Vector3<Scalar> &rpy_in,
+              const std::array<int, 3> &axes_in,
+              const EulerRotationType &rot_type_in)
+      : roll(rpy_in(0)), pitch(rpy_in(1)), yaw(rpy_in(2)), axes(axes_in),
+        rot_type(rot_type_in) {}
+
+  /**
+   * @brief Returns quaternion representation of this rotation.
+   *
+   * @return quaternion
+   */
+  Eigen::Quaternion<Scalar> ToQuaternion() const
+  {
+    return EulerAnglesToQuaternion(*this);
+  }
+
+  /**
+   * @brief Returns 3x3 rotation matrix representation of this rotation.
+   *
+   * @return 3x3 rotation matrix
+   */
+  Eigen::Matrix3<Scalar> ToRotationMatrix() const
+  {
+    return EulerAnglesToRotationMatrix(*this);
+  }
+};
+
+/**
+ * @brief Static assertion that \c EulerAngle object follows X-Y-Z Extrinsic
+ *        Euler angle convention.
+ *
+ * @tparam Scalar numeric type
+ * @param euler Euler angle object
+ */
+template<typename Scalar>
+void ASSERT_DEFAULT_EULER_CONVENTION(const EulerAngles<Scalar> &euler)
```

Comment:
This is a warning to users that currently only XYZ extrinsic exists. Should make an issue to extend Euler angle capabilities to other conventions.

Reply:
Capabilities extended in bdfce1322a036db5ad1a8e1a939c31b6042e2025